hazelcast/src/main/java/com/hazelcast/ringbuffer/Ringbuffer.java

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright (c) 2008-2015, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.ringbuffer;
+
+import com.hazelcast.core.DistributedObject;
+import com.hazelcast.core.ICompletableFuture;
+import com.hazelcast.spi.annotation.Beta;
+
+import java.util.Collection;
+import java.util.List;
+
+/**
+ * A Ringbuffer is a data-structure where the content is stored in a ring like structure. A ringbuffer has a capacity so it
+ * won't grow beyond that capacity and endanger the stability of the system. If that capacity is exceeded, than the oldest
+ * item in the ringbuffer is overwritten.
+ *
+ * The ringbuffer has 2 always incrementing sequenceId's:
+ * <ol>
+ * <li>
+ * tailSequenceId: this is the side where the youngest item is found. So the tail is the side of the ringbuffer where
+ * items are added to.
+ * </li>
+ * <li>
+ * headSequenceId: this is the side where the oldest items are found. So the head is the side where items gets
+ * discarded.
+ * </li>
+ * </ol>
+ * The items in the ringbuffer can be found by a sequenceId that is in between (inclusive) the head and tail sequenceId.
+ *
+ * If data is read from a ringbuffer with a sequenceId that is smaller than the headSequenceId, it means that the data
+ * is not available anymore and a {@link StaleSequenceIdException} is thrown.
+ *
+ * A Ringbuffer currently is not a distributed data-structure. So all data is stored in a single partition; comparable to the
+ * IQueue implementation. But we'll provide an option to partition the data in the near future.
+ *
+ * A Ringbuffer can be used in a similar way as a queue, but one of the key differences is that a queue.take is destructive,
+ * meaning that only 1 thread is able to do take an item. A ringbuffer.read is not destructive, so you can have multiple threads
+ * reading the same data.
+ *
+ * The Ringbuffer is the backing data-structure for the reliable {@link com.hazelcast.core.ITopic} implementation. See
+ * {@link com.hazelcast.config.ReliableTopicConfig}.
+ *
+ * @param <E>
+ */
+@Beta
+public interface Ringbuffer<E> extends DistributedObject {
```

Comment:
various methods talk about:
- backoff policy
- retention policy
- overflow policy. 

It would be nice to explain these concepts in a class-level JavaDoc.